ethportal-peertest/src/lib.rs

```diff
@@ -2,31 +2,85 @@ pub mod cli;
 pub mod jsonrpc;
 
 pub use cli::PeertestConfig;
-pub use jsonrpc::{get_enode, test_jsonrpc_endpoints_over_http, test_jsonrpc_endpoints_over_ipc};
+pub use jsonrpc::get_enode;
 
 use std::sync::Arc;
 
+use futures::future;
+
 use trin_core::cli::TrinConfig;
 use trin_core::jsonrpc::service::JsonRpcExiter;
 
-pub async fn launch_buddy_node() -> (String, Arc<JsonRpcExiter>) {
+pub struct PeertestNode {
+    pub enr: String,
+    pub exiter: Arc<JsonRpcExiter>,
+}
+
+pub struct AllPeertestNodes {
+    pub bootnode: PeertestNode,
+    pub nodes: Vec<PeertestNode>,
+}
+
+impl AllPeertestNodes {
+    pub fn exit(&self) {
+        self.bootnode.exiter.exit();
+        self.nodes.iter().for_each(|node| node.exiter.exit());
+    }
+}
+
+fn get_peertest_id_for_node(mut id: u8, bootnode_enr: Option<&String>) -> u16 {
+    // Peertest Id for bootnode is 1 (identified by bootnode_enr == None)
+    // For all other nodes (identified by bootnode_enr == Some()) Peertest ID needs to be
+    // incremented by one to account for the bootnode
+    if bootnode_enr.is_some() {
+        id += 1;
+    }
+    id as u16
+}
+
+pub async fn launch_node(id: u8, bootnode_enr: Option<&String>) -> anyhow::Result<PeertestNode> {
+    let id = get_peertest_id_for_node(id, bootnode_enr);
+
     // Run a client, as a buddy peer for ping tests, etc.
-    let trin_config = TrinConfig::new_from(
-        [
+    let discovery_port: u16 = 9000 + id;
```

Comment:
So the `id` here serves two purposes. 1. As the suffix on a base `9000` port. 2. As the suffix for the ipc-path (eg. `/tmp/ethportal-peertest-buddy-{id}.ipc`). I guess we could use `AtomicUsize` to generate this id / offset... but I'm not sure it's necessary? Maybe, can you just explain why you think this will be better?